core/consensus/metrics/metrics.go

```diff
@@ -0,0 +1,73 @@
+// Copyright © 2022-2024 Obol Labs Inc. Licensed under the terms of a Business Source License 1.1
+
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/obolnetwork/charon/app/promauto"
+)
+
+type ConsensusMetrics interface {
+	SetDecidedRounds(duty, timer string, rounds float64)
+	ObserveConsensusDuration(duty, timer string, duration float64)
+	IncConsensusTimeout(duty, timer string)
+	IncConsensusError()
+}
+
+var (
+	decidedRoundsGauge = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "core",
+		Subsystem: "consensus",
+		Name:      "decided_rounds",
+		Help:      "Number of rounds it took to decide consensus instances by protocol, duty and timer type.",
+	}, []string{"protocol", "duty", "timer"}) // Using gauge since the value changes slowly, once per slot.
+
+	consensusDuration = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: "core",
+		Subsystem: "consensus",
+		Name:      "duration_seconds",
+		Help:      "Duration of a consensus instance in seconds by protocol, duty and timer type.",
+		Buckets:   []float64{.05, .1, .25, .5, 1, 2.5, 5, 10, 20, 30, 60},
+	}, []string{"protocol", "duty", "timer"})
+
+	consensusTimeout = promauto.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "core",
+		Subsystem: "consensus",
+		Name:      "timeout_total",
+		Help:      "Total count of consensus timeouts by protocol, duty and timer type.",
+	}, []string{"protocol", "duty", "timer"})
+
+	consensusError = promauto.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "core",
+		Subsystem: "consensus",
+		Name:      "error_total",
+		Help:      "Total count of consensus errors by protocol",
+	}, []string{"protocol"})
+)
+
+type consensusMetrics struct {
+	protocolID string
+}
+
+func NewConsensusMetrics(protocolID string) ConsensusMetrics {
```

Comment:
Weird that golangci-lint didn't complain for missing godoc?

Reply:
It does not complain indeed. I will go over every exported function and add godoc.